docs/docsite/rst/network/user_guide/cli_parsing.rst

```diff
@@ -0,0 +1,728 @@
+   2: enp0s31f6: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc fq_codel state DOWN group default qlen 1000
+       link/ether x2:6a:64:9d:84:19 brd ff:ff:ff:ff:ff:ff
+   3: wlp2s0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
+       link/ether x6:c2:44:f7:41:e0 brd ff:ff:ff:ff:ff:ff permaddr d8:f2:ca:99:5c:82
+
+Create the native template to match this output and store it as ``templates/fedora_ip_addr_show.yaml``:
+
+.. code-block:: yaml
+
+   ---
+   - example: '1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000'
+     getval: |
+       (?x)                                                # free-spacing
+       \d+:\s                                              # the interface index
+       (?P<name>\S+):\s                                    # the name
+       <(?P<properties>\S+)>                               # the properties
+       \smtu\s(?P<mtu>\d+)                                 # the mtu
+       .*                                                  # gunk
+       state\s(?P<state>\S+)                               # the state of the interface
+     result:
+       "{{ name }}":
+           name: "{{ name }}"
+           loopback: "{{ 'LOOPBACK' in stats.split(',') }}"
+           up: "{{ 'UP' in properties.split(',')  }}"
+           carrier: "{{ not 'NO-CARRIER' in properties.split(',') }}"
+           broadcast: "{{ 'BROADCAST' in properties.split(',') }}"
+           multicast: "{{ 'MULTICAST' in properties.split(',') }}"
+           state: "{{ state|lower() }}"
+           mtu: "{{ mtu }}"
+     shared: True
+
+   - example: 'inet 192.168.122.1/24 brd 192.168.122.255 scope global virbr0'
+     getval: |
+      (?x)                                                 # free-spacing
+      \s+inet\s(?P<inet>([0-9]{1,3}\.){3}[0-9]{1,3})       # the ip address
+      /(?P<bits>\d{1,2})                                   # the mask bits
+     result:
+       "{{ name }}":
+           ip_address: "{{ inet }}"
+           mask_bits: "{{ bits }}"
+
+.. note::
+
+	 The ``shared`` key in the parser template allows the interface name to be used in subsequent parser entries. The use of examples and free-spacing mode with the regular expressions makes the template easier to read.
+
+The following example task uses ``cli_parse`` with the native parser and the example template above to parse the Linux output:
+
+.. code-block:: yaml
+
+   - name: Run command and parse
+     ansible.netcommon.cli_parse:
+       command: ip addr show
+       parser:
+         name: ansible.netcommon.native
+       set_fact: interfaces
+
+This task assumes you previously gathered facts to determine the ``ansible_distribution`` needed to locate the template. Alternately, you could provide the path in the  ``parser/template_path`` option.
+
+.. note::
+
+	The ``ansible.netcommon.native`` parsing engine is fully supported with a Red Hat Ansible Automation Platform subscription.
```

Comment:
Move information about support higher up, or maybe repeat in the list of available parsing engines? It feels a bit buried here and in the sections below about the other parsers. As a customer, I might choose only to read the detailed info for supported parsers, so making it easy to identify them is useful.